lib/qless/job.rb

```diff
@@ -234,12 +268,7 @@ def to_hash
     def requeue(queue, opts = {})
       note_state_change :requeue do
         @client.call('requeue', @client.worker_name, queue, @jid, @klass_name,
-                     JSON.dump(opts.fetch(:data, @data)),
-                     opts.fetch(:delay, 0),
-                     'priority', opts.fetch(:priority, @priority),
-                     'tags', JSON.dump(opts.fetch(:tags, @tags)),
-                     'retries', opts.fetch(:retries, @original_retries),
-                     'depends', JSON.dump(opts.fetch(:depends, @dependencies))
+                     *self.class.build_opts_array(self.enqueue_opts.merge(opts))
```

Comment:
Use `Hash#merge!` so as not to create a new Hash via `enqueue_opts` and then just throw it away via `merge`